### PR TITLE
[ISSUE #4211] Add JavaDoc for eventmesh.runtime.admin APIs (not in dashboard)

### LINKED ISSUE #4211: [Doc] Add JavaDoc for eventmesh.runtime.admin APIs (not in dashboard)
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Documentation Related

This is a subtask of GLCC Subject https://github.com/apache/eventmesh/issues/4040.

Write comments for classes and methods in the `org.apache.eventmesh.runtime.admin` package which haven't been integrated in the eventmesh-dashboard.

~~WIP.~~

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Pil0tXia**
Fixes #4211.

### Motivation

Please see https://github.com/apache/eventmesh/issues/4211.

### Modifications

Write comments for classes and methods in the `org.apache.eventmesh.runtime.admin` package which haven't been integrated in the eventmesh-dashboard.

Some webhook related classes are also commented in order to provide help for others to understand, such as the logic of which implementation to use for `WebHookConfigOperation` interface.

Example:

image

No changes are made in codes except the redundant `toString()` is removed in `RedirectClientByIpPortHandler`.

### Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Review by mxsm — COMMENTED** [review]
@Pil0tXia IMO,Both methods and classes can be accompanied by explanatory comments, but class properties and method code do not require excessive comments. Comments serve the purpose of enhancing code comprehension. Regarding class properties, if the property names are semantically clear and self-explanatory, there is no need to add comment explanations. For complex logic, adding appropriate comments can be helpful, while for relatively simple logic, comments may not be necessary.   

One more small suggestion, try to compress multiple commits into one for a Pull Request (PR). Keeping the commit messages concise and focused helps other developers better understand the code evolution.

**Comment by Pil0tXia** [comment]
@mxsm Thank you for your valuable opinion. For the existing single-line comments in the code, I hope they can be presented in natural language to reduce the time others spend on inspecting the source code. This way, one can more directly understand a relatively complex data structure and its purpose for being used here.

For instance, in this picture, the specific meaning of traversing keys and values in nested iterations requires looking at the property definitions of the entity class to understand. However, single-line comments can precisely indicate different levels and directly inform readers about the relationship between these entity classes and the concept of "client group."

image

By doing so, readers only need to go through the comments to understand the workflow of this handler and to clearly divide sections for retrieval, validation, processing, and response logic.

Of course, I will also consider your suggestions to further streamline single-line comments with relatively clear meanings. 

As for what you mentioned about "class properties", if you are referring to the attributes of entity classes, rest assured that I have hardly added comments for class properties, except for improving the existing ones to better express their purposes.

Lastly, I will squash multiple commits into a new branch for the pull request. Thank you for your excellent suggestions.

**Comment by Pil0tXia** [comment]
@mxsm I have streamlined and reduced existing single-line comments. 😉

**Comment by Pil0tXia** [comment]
@mxsm @pandaapo @xwm1992 PTAL~😊

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/DeleteWebHookConfigHandler.java [hunk 1]
```diff
@@ -33,29 +33,69 @@
...
+ * {@linkplain org.apache.eventmesh.webhook.admin.FileWebHookConfigOperation#deleteWebHookConfig FileWebHookConfigOperation}
+ * method as implementation to delete the WebHook configuration file;
+ * <p>
+ * When {@code eventMesh.webHook.operationMode=nacos}, It calls the
+ * {@linkplain org.apache.eventmesh.webhook.admin.NacosWebHookConfigOperation#deleteWebHookConfig NacosWebHookConfigOperation}
+ * method as implementation to delete the WebHook configuration from Nacos.
+ * <p>
+ * The {@linkplain org.apache.eventmesh.webhook.receive.storage.HookConfigOperationManager#deleteWebHookConfig HookConfigOperationManager}
+ * which implements the {@linkplain org.apache.eventmesh.webhook.api.WebHookConfigOperation WebHookConfigOperation}
+ * interface, does not participate in the implementation of this endpoint.
```
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/DeleteWebHookConfigHandler.java:55]: The reason I specifically emphasize the difference between `HookConfigOperationManager` and `FileWebHookConfigOperation` and `NacosWebHookConfigOperation` is that I spent quite some time understanding the distinctions among these three. 

When you view the implementation of the WebHookConfigOperation interface in IDEA, you will see these three on the screenshot, which might give the impression that there are three different implementations of this method. However, that is not the case in reality. In the `/webhook` endpoints, EventMesh only utilizes the `file` and `nacos` implementations, and `HookConfigOperationManager` is used differently from both of these and only one method is being used, which can be somewhat confusing and misleading. 

Therefore, I think it's necessary to provide an explanation. Perhaps stating in the javadoc that "they are different" is not enough; I need to clarify "why they are different".

image

[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/DeleteWebHookConfigHandler.java]: Resolved in https://github.com/apache/eventmesh/pull/4272/commits/92228248444eb344b38b09c673d6b9d5728b6d04.

File: eventmesh-webhook/eventmesh-webhook-receive/src/main/java/org/apache/eventmesh/webhook/receive/storage/HookConfigOperationManager.java [hunk 2]
```diff
@@ -42,6 +41,20 @@
...
+/**
+ * This class manages the operations related to WebHook configurations. Mainly used by
+ * {@linkplain org.apache.eventmesh.webhook.receive.WebHookController#execute WebHookController}
+ * to retrieve existing WebHook configuration by callback path when processing received WebHook data from manufacturers.
+ * <p>
+ * This class is initialized together with the {@linkplain org.apache.eventmesh.webhook.receive.WebHookController WebHookController}
+ * during the initialization phase of the {@code EventMeshHTTPServer}.
+ * Differs from the other two implementations that implement the
+ * {@linkplain org.apache.eventmesh.webhook.api.WebHookConfigOperation WebHookConfigOperation} interface,
+ * which are located in the {@code org.apache.eventmesh.webhook.admin} package.
```
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-webhook/eventmesh-webhook-receive/src/main/java/org/apache/eventmesh/webhook/receive/storage/HookConfigOperationManager.java]: Resolved in https://github.com/apache/eventmesh/pull/4272/commits/0890078361992463b616a9e002006fd58fb04cc1 (Explain HookConfigOperationManager more clearly in WebHookConfigOperation).

I have reworked this JavaDoc and moved it to `WebHookConfigOperation` interface for better demonstration.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/DeleteWebHookConfigHandler.java [hunk 3]
```diff
@@ -33,29 +33,69 @@
...
     public DeleteWebHookConfigHandler(WebHookConfigOperation operation, HttpHandlerManager httpHandlerManager) {
         super(httpHandlerManager);
         this.operation = operation;
     }
 
-
+    /**
+     * Handles the HTTP requests by deleting a WebHook configuration.
+     * <p>
+     * This method is an implementation of {@linkplain com.sun.net.httpserver.HttpHandler#handle(HttpExchange)  HttpHandler.handle()}.
```
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/DeleteWebHookConfigHandler.java]: Resolved in https://github.com/apache/eventmesh/pull/4272/commits/18fa9ac7adc96c973eb95dd9d6fb9de535fdfaed (Concise the implement of handle())

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/QueryRecommendEventMeshHandler.java [hunk 4]
```diff
@@ -38,30 +38,61 @@
...
+     * This method is an implementation of {@linkplain com.sun.net.httpserver.HttpHandler#handle(HttpExchange)  HttpHandler.handle()}.
+     *
+     * @param httpExchange the exchange containing the request from the client and used to send the response
+     * @throws IOException if an I/O error occurs while handling the request
+     */
     @Override
     public void handle(HttpExchange httpExchange) throws IOException {
         String result = "";
         try (OutputStream out = httpExchange.getResponseBody()) {
+            // Check whether the registry is enabled
```
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/QueryRecommendEventMeshHandler.java]: For now, I am very familiar with this portion of the code, and I can understand it without comments. However, for new contributors, these comments will undoubtedly be helpful. The comments provide clearer delineation of the functionality in long stretches of code, making it more pleasing to the eye. 

Nevertheless, I have removed some relatively simple comments in https://github.com/apache/eventmesh/pull/4272/commits/b2d4ca8118b2d53875f76cd8203d0794debf3fbc.

**Review by mxsm — APPROVED** [review]
LGTM

**Pull request merged**